sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/subquery.scala

```diff
@@ -105,6 +105,13 @@ object PredicateSubquery {
       case _ => false
     }.isDefined
   }
+  def hasNullAwarePredicate(e: Expression): Boolean = {
+    e.find(_.isInstanceOf[Not]).isDefined &&
```

Comment:
This could have false positive (the join could be not-null-aware, but we think it's null-awear), that's OK for now.